packages/go/cypher/models/pgsql/translate/expansion.go

```diff
@@ -712,6 +696,111 @@ func (s *Translator) buildExpansionPatternRoot(part *PatternPart, traversalStep
 	return expansion.Build(traversalStep.Expansion.Value.Binding.Identifier), nil
 }
 
+type ExpansionComponents struct {
+	TerminalNode         *BoundIdentifier
+	RootNode             *BoundIdentifier
+	RightNodeConstraints *Constraint
+	EdgeConstraints      *Constraint
+	LeftNodeConstraints  *Constraint
+	PrimerVisible        *pgsql.IdentifierSet
+	RecursiveVisible     *pgsql.IdentifierSet
+	PrimerWhereClause    pgsql.Expression
+	RecursiveWhereClause pgsql.Expression
+}
```

Comment:
It was mentioned during live review that this opens us up to a lot of nil pointers depending on use case. What mitigation strategies can we pursue until a more elegant structure presents itself? It seems like at a minimum, this type could be private as it's not used outside this package and given the volatile nature of it, probably shouldn't be used outside of this file

Reply:
Patching this up.